core/api/service/api_service.hpp

```diff
@@ -0,0 +1,48 @@
+/**
+ * Copyright Soramitsu Co., Ltd. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#ifndef KAGOME_CORE_API_SERVICE_HPP
+#define KAGOME_CORE_API_SERVICE_HPP
+
+#include <functional>
+
+#include "api/transport/listener.hpp"
+
+namespace kagome::api {
+
+  class JRPCProcessor;
+
+  class ApiService : public std::enable_shared_from_this<ApiService> {
+   public:
+    template <class T>
+    using sptr = std::shared_ptr<T>;
+
+    /**
+     * @brief constructor
+     * @param context reference to io context
+     * @param listener sptr to listener instance
+     * @param processor sptr to json-rpc processor instance
+     */
+    ApiService(std::shared_ptr<Listener> listener,
+               std::shared_ptr<JRPCProcessor> processor);
+
+    virtual ~ApiService() = default;
+    /**
+     * @brief starts service
+     */
+    virtual void start();
+
+    /**
+     * @brief stops service
+     */
+    virtual void stop();
```

Comment:
Why are the methods virtual?

Reply:
Refactored interface, in current implementation stop is not required to be virtual, instead made respond() virtual and that simplified many things.